test/server/cards/02_SHD/upgrades/Foundling.spec.js

```diff
@@ -0,0 +1,39 @@
+describe('Foundling\'s', function () {
+    integration(function () {
+        describe('Foundling', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['foundling', 'mandalorian-warrior'],
+                        groundArena: ['battlefield-marine', 'pyke-sentinel', 'protector-of-the-throne'],
+                    },
+                    player2: {
+                        groundArena: ['snowspeeder']
+                    }
+                });
+            });
+
+            it('ability should give the attached card the Mandalorian trait', function () {
+                // check battlefield marine trait
+                expect(this.battlefieldMarine.hasSomeTrait('mandalorian')).toBeFalse();
+
+                // add battlefield marine foundling
+                this.player1.clickCard(this.foundling);
+                expect(this.player1).toBeAbleToSelectExactly([this.battlefieldMarine, this.snowspeeder, this.pykeSentinel, this.protectorOfTheThrone]);
+                this.player1.clickCard(this.battlefieldMarine);
+
+                // check players units traits
+                expect(this.battlefieldMarine.hasSomeTrait('mandalorian')).toBeTrue();
```

Comment:
For consistency, let's also use `Trait.Mandalorian` throughout the tests. It gets really messy if we use strings instead of the enum name.